### PR TITLE
GH-259: move cms to core

### LINKED ISSUE #259: Move cms store to core library
The cms store is wired into the component library. As part of GH-250, we move ngrx store outside the component library and only interact with the facades.

To do:
- follow checklist in GH-250
- have a look at GH-257 for an example
- should be fine but if needed we can have a kickoff with tobias and/or stan

Dependency: Make sure the related facades #230 #322 #157 are done.

Acceptance criteria: 
- it builds :)
- everything in checklist done

### PR DESCRIPTION
**Pull request by RomeroDiver**
Closes #259

### CODE REVIEW

File: projects/core/src/cms/store/cms-store.module.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import { metaReducers } from './reducers/index';
+import { CmsOccModule } from '../occ/cms-occ.module';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    HttpClientModule,
+    CmsOccModule,
+    StoreModule.forFeature(CMS_FEATURE, reducerToken, { metaReducers }),
```
Comment (dunqan) [inline projects/core/src/cms/store/cms-store.module.ts]: You should probably also import StateModule here, to make it complete/standalone

File: projects/core/src/cms/store/reducers/page.reducer.ts [hunk 2]
```diff
@@ -1,11 +1,6 @@
 import * as fromPageData from '../actions/page.action';
-import { Page } from '../../models/page.model';
-
-export interface PageState {
-  entities: { [context: string]: Page };
-  count: number;
-  latestPageKey: string;
-}
+import { Page } from '../../model/page.model';
+import { PageState } from '../cms-state';
```
Comment (dunqan) [inline projects/core/src/cms/store/reducers/page.reducer.ts]: Please move selectors at the end of reducers to selectors (where memoized selectors are defined)

File: projects/core/src/cms/store/selectors/navigation-entry-item.selectors.ts [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
+import * as fromFeature from '../reducers/index';
+import { CmsState, TestState } from '../cms-state';
+
+export const getNavigationEntryItemsSelector = (state: TestState) =>
+  state.nodes;
 
 export const getNavigationEntryItemState: MemoizedSelector<
   any,
-  fromNavigationEntryItem.NavigationItemState
+  TestState
```
Comment (dunqan) [inline projects/core/src/cms/store/selectors/navigation-entry-item.selectors.ts]: What is TestState?

File: projects/core/src/cms/store/selectors/navigation-entry-item.selectors.ts [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
-import * as fromFeature from '../reducers';
-import * as fromNavigationEntryItem from '../reducers/navigation-entry-item.reducer';
+import * as fromFeature from '../reducers/index';
+import { CmsState, TestState } from '../cms-state';
+
+export const getNavigationEntryItemsSelector = (state: TestState) =>
+  state.nodes;
 
 export const getNavigationEntryItemState: MemoizedSelector<
   any,
```
Comment (dunqan) [inline projects/core/src/cms/store/selectors/navigation-entry-item.selectors.ts]: This any could be replaced by StateWithCms

**Comment by RomeroDiver** [comment]
The build is not passing actually - there is one place where it fails, I've added a TestState there in order to pass it. The test state differs from the actual one only with the nodes type

File: projects/core/src/cms/occ/component-mapper.service.ts [hunk 5]
```diff
@@ -47,12 +47,6 @@ export class ComponentMapperService {
     if (!componentConfig) {
       if (this.missingComponents.indexOf(typeCode) === -1) {
         this.missingComponents.push(typeCode);
-        console.warn(
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/cms/occ/component-mapper.service.ts]: 👍

**Pull request merged**